File: backend/app/schemas/voice.py
import uuid

from pydantic import BaseModel


class VoiceOut(BaseModel):
    id: uuid.UUID
    name: str
    preset_key: str
    language_code: str | None
    similarity: float
    stability: float
    style: float
    speed: float
    sample_audio_url: str | None
    is_active: bool

    model_config = {"from_attributes": True}


class VoicePreviewResponse(BaseModel):
    voice_id: uuid.UUID
    audio_base64: str
    content_type: str = "audio/mpeg"

File: frontend/lib/types/voice.ts
export interface Voice {
  id: string;
  name: string;
  preset_key: string;
  language_code: string | null;
  similarity: number;
  stability: number;
  style: number;
  speed: number;
  sample_audio_url: string | null;
  is_active: boolean;
}

export interface VoicePreviewResponse {
  voice_id: string;
  audio_base64: string;
  content_type: string;
}
